langchain/vectorstores/pinecone.py

```diff
@@ -95,6 +95,14 @@ def add_texts(
         self._index.upsert(vectors=docs, namespace=namespace, batch_size=batch_size)
         return ids
 
+    def _similarity_search_with_relevance_scores(
+        self,
+        query: str,
+        k: int = 4,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        return self.similarity_search_with_score(query, k=k)
```

Comment:
`_similarity_search_with_relevance_scores` is meant to return normalized scores where 0 indicates dissimilarity and 1 indicates similarity. are we sure `similarity_search_with_score` returns scores like this? I think it might be returning some distance metric, which would mean directionality is backwards

Reply:
I am using this in a project at the moment and it is working perfectly!
I am using the embedding model text-embedding-ada-002. 

Below is snippet from my codebase. 
Please let me know if this is sufficient. 
```
model_name = "text-embedding-ada-002"

embed = OpenAIEmbeddings(
    model=model_name,
    openai_api_key="..."
)

# switch back to normal index for langchain
index = pinecone.Index("groceries")

vectorstore = Pinecone(
    index, embed.embed_query, text_field
)

llm = ChatOpenAI(
    openai_api_key="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",
    model_name='gpt-3.5-turbo',
    verbose=True,
    temperature=0.0
)

qa = RetrievalQA.from_chain_type(
    llm=llm,
    chain_type="stuff",
    verbose=True,
    chain_type_kwargs={"prompt": PROMPT},
    retriever=vectorstore.as_retriever(
    search_type="similarity_score_threshold",
    search_kwargs={"score_threshold": 0.8})
)

```

Reply:
@murphp15, can you confirm that the scores are normalized to [0, 1]? It looks like that's what's holding this up from approval and release — and it would be great functionality to have! Appreciate your work to make it happen. Thanks! (I know you said it's working for you, but that's not quite the same as confirming that the scores are always normalized to [0, 1] and thus will work for others too.)

Reply:
If you are using cosine similarity then I can guarantee it is between 0 and 1.

It has the same characteristics as the implementations for the other vectorStores. 

@dev2049 Was concerned that it was backwards. That is certainly not the case, so in my opinion all is good. 



@chrislrobert Please let me know if this is a sufficient answer for you?

Reply:
Thanks, @murphp15, for the quick clarification! I'm hoping that will be enough for @dev2049 or @hwchase17 to review and possibly approve for release. I'd love to begin relying on this capability ASAP!